File: Gixor-Backend/Gixor-Api/Gixor-Api-GitHub/src/main/java/com/cpy3f2/Gixor/Domain/Discussion.java
package com.cpy3f2.Gixor.Domain;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

/**
 * @author : simon
 * @description :
 * @last : 2024-11-05 21:47
 * Copyright (c) 2024. 保留所有权利。
 * @since : 2024-11-05 21:47
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@JsonIgnoreProperties(ignoreUnknown = true)
public class Discussion {
    private String title;
    private String body;

    @JsonProperty("created_at")
    private LocalDateTime createdAt;

    @JsonProperty("author")
    private String authorName;


    @JsonProperty("category")
    private String categoryName;


    private List<Comment> comments;

    @Data
    @Builder
    @JsonIgnoreProperties(ignoreUnknown = true)
    public static class Comment {
        private String body;
        private String authorName;
    }
}

File: Gixor-Mobile/app/src/main/java/com/cpy3f2/gixor_mobile/model/entity/Discussion.kt
package com.cpy3f2.gixor_mobile.model.entity

import java.time.LocalDateTime
import com.google.gson.annotations.SerializedName;

/**
 * @author : lya
 * @description :讨论
 * @last : 2024-11-05 21:47
 * Copyright (c) 2024. 保留所有权利。
 * @since : 2024-11-05 21:47
 */
data class Discussion(
    val title: String,
    val body: String,

    @SerializedName("created_at")
    val createdAt: LocalDateTime,

    @SerializedName("author")
    val authorName: String,

    @SerializedName("category")
    val categoryName: String,

    val comments: MutableList<Comment>
) {
    data class Comment(
        val body: String,
        val authorName: String
    )
}
